Sources/GRPC/Concurrency/AsyncWriter.swift

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2021, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if compiler(>=5.5)
+import NIOConcurrencyHelpers
+import NIOCore
+
+/// An asynchronous writer which forwards messages to a delegate.
+///
+/// Forwarding of messages to the delegate may be paused and resumed (by calling ``pause()``
+/// and ``resume()``, respectively). When the writer is paused calls to ``write(_:)`` may suspend.
+/// Any calls which are suspended may be continued at later (after ``resume()`` has been called).
+///
+/// The writer must also be "finished" with a final value: as for writing, calls to ``finish(_:)``
+/// may suspend if the writer has been paused.
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+internal struct AsyncWriter<
+  Element,
+  End,
+  Delegate: AsyncWriterDelegate
+> where Delegate.Element == Element, Delegate.End == End {
```

Comment:
```suggestion
internal struct AsyncWriter<Delegate: AsyncWriterDelegate> {
```

And then just use `Delegate.Element` and `Delegate.End` in the function signatures.

Reply:
This makes stuff IMHO easier to read. Since it shows right away where this `End` and `Element` comes from